src/dynarec/dynarec_arm_660f.c

```diff
@@ -341,7 +341,19 @@ uintptr_t dynarec660F(dynarec_arm_t* dyn, uintptr_t addr, uintptr_t ip, int nins
                     VUZPQ_16(q0, v1);
                     VSUBQ_16(q0, q0, v1);
                     break;
-
+                case 0x06:
+                    INST_NAME("PHSUBD Gx, Ex");
+                    nextop = F8;
+                    GETGX(q0, 1);
+                    GETEX(q1, 0);
+                    if ((nextop & 0xDA) == 0xDA) {
```

Comment:
Why 0xDA?

Reply:
according 66 0F 38 05：

Reply:
case 0x05:
                    INST_NAME("PHSUBW Gx, Ex");
                    nextop = F8;
                    GETGX(q0, 1);
                    GETEX(q1, 0);
                    if((nextop&0xC0) == 0xC0) {
                        v1 = fpu_get_scratch_quad(dyn);
                        VMOVQ(v1, q1);
                    } else
                        v1 = q1;
                    VUZPQ_16(q0, v1);
                    VSUBQ_16(q0, q0, v1);
                    break;

Reply:
it had judge nextop ,so I do it too.

Reply:
use this test, it successed!
#include <stdio.h>

int b[4] = {0, 0, 0, 0};

int main(void)
{
    int a[4] = {0x12345678, 0x9abcdef0, 0x56789abc, 0xdef01234};
    int c[4] = {0x9922aabb, 0x88332520, 0x1278990c, 0xbbccddaa};
    int *p = &b[0];

    __asm__ __volatile__ ("movd %0, %%xmm0\r\n"
                          "movd %1, %%xmm1\r\n"
                          "punpckldq %%xmm0, %%xmm1\r\n"
                          "movd %2, %%xmm0\r\n"
                          "movd %3, %%xmm2\r\n"
                          "movdqa %%xmm2, %%xmm3\r\n"
                          "punpckldq %%xmm0, %%xmm3\r\n"
                          "movdqa %%xmm3, %%xmm0\r\n"
                          "punpcklqdq %%xmm1, %%xmm0\r\n"
                          ::"r"(a[0]),"r"(a[1]),"r"(a[2]),"r"(a[3]));

    __asm__ __volatile__ ("movd %0, %%xmm1\r\n"
                          "movd %1, %%xmm2\r\n"
                          "punpckldq %%xmm1, %%xmm2\r\n"
                          "movd %2, %%xmm1\r\n"
                          "movd %3, %%xmm3\r\n"
                          "movdqa %%xmm3, %%xmm4\r\n"
                          "punpckldq %%xmm1, %%xmm4\r\n"
                          "movdqa %%xmm4, %%xmm1\r\n"
                          "punpcklqdq %%xmm2, %%xmm1\r\n"
                          ::"r"(c[0]),"r"(c[1]),"r"(c[2]),"r"(c[3]));

    __asm__ __volatile__ ("phsubd %xmm0, %xmm1\r\n");

    __asm__ __volatile__ ("movdqa %%xmm1, (%0)\r\n"::"r"(p));

#if 1
    if((b[0] == 0xa954449e) && (b[1] == 0xef107a65)
       && (b[2] == 0x88777778) && (b[3] == 0x88888878))
    {
        printf("success!\n");
    }
    else
    {
        printf("wrong!\n");
    }
#endif

    return 0;
}

Reply:
I understand the 0xC0 judge, I will pull request later.
<img width="520" alt="14008cd2-709a-4622-8494-b13faf987862" src="https://user-images.githubusercontent.com/76616478/127966448-e5d72136-733a-42df-9d8d-2f5db767f02d.png">

Reply:
Yes. C0 mask is the "register" part of Mod R/M. This needs to be tested for that opcode because the VUNZP whill change it, so it need to be copied in a temporary register first. If the Mod R/M is a memory, then the register is already temporary.